File: problems/10_backtracking/0078_subsets/solution.swift
// Problem: 0078. Subsets
// Difficulty: Medium

import Foundation

class Solution {
    func subsets(_ nums: [Int]) -> [[Int]] {
        // TODO: implement
        return []
    }
}

func normalize(_ xs: [[Int]]) -> [[Int]] {
    // TODO: implement
    return []
}

func checkSets(_ a: [[Int]], _ e: [[Int]], _ msg: String) {
    let na = normalize(a), ne = normalize(e)
    if na != ne { fatalError("FAIL \(msg): \(na) vs \(ne)") }
}

// MARK: - Tests
// let s = Solution()

// Test 1
// let result = s.solve(...)

print("All tests passed!")

File: problems/10_backtracking/0078_subsets/solution.py
"""
Problem: 0078. Subsets
Difficulty: Medium
URL: https://leetcode.com/problems/subsets/

解法：回溯選或不選；時間 O(N*2^N)，空間 O(N)。
"""

from __future__ import annotations

from typing import List

class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        pass


def _norm(xs):
    return sorted([sorted(s) for s in xs])

if __name__ == "__main__":
    solution = Solution()

    # Test 1
    # result = solution.solve(...)
    # assert result == expected

    print("All tests passed!")
